site-builder/src/pages/[slug].astro

```diff
@@ -0,0 +1,93 @@
+---
+import { expect } from "utils/assert";
+
+import type { FileUpload } from "../layouts/Layout.astro";
+import PubAbstract from "../components/PubAbstract.astro";
+import PubContent from "../components/PubContent.astro";
+import PubHeader from "../components/PubHeader.astro";
+import PubSidebar from "../components/PubSidebar.astro";
+import Layout from "../layouts/Layout.astro";
+import { getHeader, getJournal, getJournalArticles, getPages } from "../lib/client/queries";
+import { getFieldValue } from "../lib/getValue";
+
+const { slug } = Astro.params;
+
+export async function getStaticPaths() {
+	const [pages, journal, header] = await Promise.all([
+		getPages(),
+		getJournal({
+			depth: 1,
+			withRelatedPubs: true,
+		}),
+		getHeader(),
+	]);
+
+	const paths = pages
+		.map((page) => {
+			const slug = page.values?.find((value) => value.fieldName === "Slug")?.value;
+			if (!slug) {
+				console.log(page);
+				return;
+			}
+			const layout = page.values?.find((value) => value.fieldName === "Layout")?.value;
+			if (!layout) {
+				console.log(page);
+				return;
+			}
+
+			const isPublic = page.values?.find((value) => value.fieldName === "Is Public")?.value;
+			if (!isPublic) {
+				return;
+			}
+
+			return {
+				params: {
+					slug,
+				},
+				props: { page, journal, header },
+			};
+		})
+		.filter((page) => !!page);
+
+	return paths;
+}
```

Comment:
i really like this about astro, we only need to fetch that data once and then can use it for each page!

however, we might need to do it differently: if you have a sufficiently large community this requires keeping every pub in memory (i think) in order to build the entire site, which is will obviously take it down very quickly.

instead we should probably write the results to disk or something (stream it ideally) and then read it for each page. thisll be slower but easier to scale.

ive tested the same approach with `next`, which is much slower for doing this (20sec builds vs 2s on astro)